### PR TITLE
enhance(navigate): make wikilinks clickable in hover

### PR DESCRIPTION
**Pull request by jonathanyeung**
## enhance(navigate): make wikilinks clickable in hover

Makes any wikilinks in hover clickable, which will navigate to the note via `GoToNote`

Coauthored by @SeriousBug.

---

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [ ] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [ ] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [ ] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [ ] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [ ]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/engine-server/src/markdown/remark/dendronPreview.ts [hunk 1]
```diff
@@ -30,6 +30,32 @@ export function makeImageUrlFullPath({
   node.url = fpath;
 }
 
+/**
+ * Transforms any wiklinks into a vscode command URI for gotoNote.
+ */
+function commandUriForWikilink({
```
Comment (SeriousBug) [inline packages/engine-server/src/markdown/remark/dendronPreview.ts:36]: nitpick: the name sounds like it returns the command URI, when it actually sets it

**Review by SeriousBug — APPROVED** [review]
Looks good to me, and works well in manual testing. Just left one nitpick.

**Review by Harshita-mindfire — APPROVED** [review]
Lgtm, works well in manual testing.

File: packages/engine-server/src/markdown/remark/dendronPreview.ts (hunk 1 above)
Comment (jonathanyeung) [reply packages/engine-server/src/markdown/remark/dendronPreview.ts]: amended to `modifyWikilinkValueToCommandUri()`
